Store/DataStoreModule.py

```diff
@@ -52,12 +52,28 @@ def __init__(self, db_username, db_password, db_host, db_port, db_name, db_type=
         # TEMP list of values for defaulted IDs, to be replaced by missing info lookup mechanism
         self.defaultUserId = 1  # DevUser
 
-        if db_type == 'sqlite':
+        # Instance attributes which are necessary for initialise method
+        self.db_name = db_name
+        self.db_type = db_type
+
+    def initialise(self):
+        """Create schemas for the database"""
+
+        Base.metadata.bind = self.engine
+        if self.db_type == 'sqlite':
             try:
                 # Attempt to create schema if not present, to cope with fresh DB file
                 Base.metadata.create_all(self.engine)
             except OperationalError:
-                print("Error creating database schema, possible invalid path? ('" + db_name + "'). Quitting")
+                print("Error creating database schema, possible invalid path? ('" + self.db_name + "'). Quitting")
+                exit()
+        elif self.db_type == 'postgres':
+            try:
+                #  ensure that create schema scripts created before create table scripts
+                event.listen(Base.metadata, 'before_create', CreateSchema('datastore_schema'))
+                Base.metadata.create_all(self.engine)
+            except OperationalError:
+                print("Error creating database schema, possible invalid path? ('" + self.db_name + "'). Quitting")
```

Comment:
The error message about having an invalid path is only relevant for Sqllite, if there is an error with Postgres setting up the schema I suppose it could be a range of things, but the db_name field will really be a database name, not a path as it is for Sqlite

Reply:
generic error message added for Postgres:
https://github.com/debrief/python-import-library/pull/8/commits/92f81ced44838f1a724bd1a1726ea001ca8e8173